vm/zstd.hpp

```diff
@@ -0,0 +1,7 @@
+// Copyright (C) 2022-2024 nomennescio
+// See https://factorcode.org/license.txt for BSD license.
+
+namespace lib { namespace zstd { extern "C" {
```

Comment:
why does it need a C++ namespace instead of just including ``zstd.c`` in image.cpp?

Reply:
Because zstd.c -- which is generated externally and I don't want to edit -- by default put everything in the global namespace, which can lead to name clashes. This prevents any (future) name clash.

Reply:
And to keep compilation snappy, I prefer to prevent unnecessary source dependencies; `zstd.o` won't change until a new version of the lib is generated. (In that sense it's a bit unfortunate Factor uses one `master.hpp` entangling all dependencies.)